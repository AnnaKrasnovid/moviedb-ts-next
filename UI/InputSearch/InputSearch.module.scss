@import '../../styles/variables.scss';

@keyframes search-magnifier {
    from {
        transform: scale(1, 1);
    }

    to {
        transform: scale(1.15, 1.15);
        opacity: 1;
    }
}

.input-box {
    width: 100%;

    &__box {
        padding: 0 16px;
        display: grid;
        grid-template-columns: 1fr max-content;
        align-items: center;
        gap: 16px;
        box-sizing: border-box;
        border: 1px solid #424242;
        background-color: #1B1E1E;
        border-radius: 50px;
        box-shadow: 0px 7px 16px rgba(0, 0, 0, 0.2);
        backdrop-filter: blur(20px);

        &_active {
            background: rgba(28, 29, 31, 0.5);
            border: 1px solid #7c7b7b;
            box-shadow: 0px 7px 16px rgba(0, 0, 0, 0.5);
            backdrop-filter: blur(30px);
        }
    }

    &__input {
        padding: 12px 0 10px;
        background-color: #1B1E1E;
        display: block;
        border: 0;
        outline: none;
        width: 100%;
        height: 36px;
        border-radius: 50px;
        font-weight: 400;
        font-size: 14px;
        line-height: 40px;
        color: #FFFFFF;

        &::placeholder {
            font-weight: 400;
            font-size: 14px;
            line-height: 21px;
            color: #8B8B8B;
        }
    }

    &__icon {
        padding: 0;
        background-image: url(../../images/loupe.svg);
        background-color: #1B1E1E;
        background-repeat: no-repeat;
        background-position: center;
        cursor: pointer;
        opacity: 0.5;
        height: 24px;
        width: 24px;
        border: 0;

        &_active {
            animation-name: search-magnifier;
            animation-delay: 0;
            animation-duration: 1.5s;
            animation-fill-mode: forwards;
            animation-iteration-count: infinite;
            animation-timing-function: ease-in-out;
        }
    }
}