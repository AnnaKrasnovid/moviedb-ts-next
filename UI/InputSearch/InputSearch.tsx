import { memo, useState } from 'react';

import Error from '../Error/Error';

import { InputSearchInt } from '../../settings/interfaces';

import styles from './InputSearch.module.scss';

function InputSearch({searchValue,setSearchValue }:InputSearchInt) {
    const [isActiveSearchClass, setIsActiveSearchClass] = useState(false);
    const [isActiveInput, setIsActiveInput] = useState(false);
    
    function handleMouseEnter() {
        setIsActiveSearchClass(true);
    }

    function handleMouseLeave() {
        setIsActiveSearchClass(false);
    }

    function handleOnFocusInput() {
        setIsActiveInput(true);
    }

    function handleOnBlurInput() {
        setIsActiveInput(false);
    }

    function handleChange(e: any) {
        setSearchValue(e.target.value)
    }
    
    return (
        <div
            className='input-box'
            onMouseEnter={handleMouseEnter}
            onMouseLeave={handleMouseLeave}
            onFocus={handleOnFocusInput}
            onBlur={handleOnBlurInput}>
            <div className={`${styles['input-box__box']} ${isActiveSearchClass || isActiveInput ? styles['input-box__box_active'] : ''}`}>
                <input
                    id='search'
                    name='search'
                    className={styles['input-box__input']}
                    type='text'
                    placeholder='Поиск'
                    minLength={1}
                    autoComplete='off'
                    value={searchValue}
                    onChange={ handleChange}
                />
                <span className={`${styles['input-box__icon']} ${isActiveInput ? styles['input-box__icon_active'] : ''}`} ></span>
            </div>
            <Error text='' />
        </div>
    )
}

export default memo(InputSearch);