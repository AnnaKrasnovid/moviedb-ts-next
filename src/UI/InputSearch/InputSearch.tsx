import { memo, useCallback, useState } from 'react';

import Error from '@/UI/Error/Error';

import { InputSearchInt } from '@/settings/interfaces';

import styles from './InputSearch.module.scss';

function InputSearch({searchValue,setSearchValue }:InputSearchInt) {
    const [isActiveSearchClass, setIsActiveSearchClass] = useState(false);
    const [isActiveInput, setIsActiveInput] = useState(false);
    
    const handleMouseEnter=useCallback(()=> {
        setIsActiveSearchClass(true);
    },[isActiveSearchClass])

    const handleMouseLeave=useCallback(()=>{
        setIsActiveSearchClass(false);
    },[isActiveSearchClass])

    const handleOnFocusInput=useCallback(()=>{
        setIsActiveInput(true);
    },[isActiveInput])

    const  handleOnBlurInput=useCallback(()=>{
        setIsActiveInput(false);
    },[isActiveInput])

    function handleChange(e: any) {
        setSearchValue(e.target.value)
    }
    
    return (
        <div
            className='input-box'
            onMouseEnter={handleMouseEnter}
            onMouseLeave={handleMouseLeave}
            onFocus={handleOnFocusInput}
            onBlur={handleOnBlurInput}
            >
            <div className={`${styles['input-box__box']} ${isActiveSearchClass || isActiveInput ? styles['input-box__box_active'] : ''}`}>
                <input
                    id='search'
                    name='search'
                    className={styles['input-box__input']}
                    type='text'
                    placeholder='Поиск фильма'
                    minLength={1}
                    autoComplete='off'
                    value={searchValue}
                    onChange={ handleChange}
                />
                <span className={`${styles['input-box__icon']} ${isActiveInput ? styles['input-box__icon_active'] : ''}`} ></span>
            </div>
            <Error text='' />
        </div>
    )
}

export default memo(InputSearch);