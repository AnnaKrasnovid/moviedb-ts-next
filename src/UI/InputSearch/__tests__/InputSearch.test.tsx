import '@testing-library/jest-dom';
import { render, screen } from "@testing-library/react";

import InputSearch from '../InputSearch';
import userEvent from '@testing-library/user-event';

const onChange = jest.fn();

describe('InputSearch', () => {
    it('Рендер компонента', () => {
        render(
            <InputSearch
                searchValue=''
                setSearchValue={onChange}
            />
        );

        const input = screen.getByRole('textbox');
        expect(input).toBeInTheDocument();
    });

    it('добавить текст в инпут', async () => {
        render(
            <InputSearch
                searchValue=''
                setSearchValue={onChange}
            />
        );

        const input = screen.getByRole('textbox');
        await userEvent.type(input, 'фильм');
        expect(onChange).toHaveBeenCalledTimes(5);
    });

    it('рендер комполента с value !== пустой строке', () => {
        render(
            <InputSearch
                searchValue='сериал'
                setSearchValue={onChange}
            />);

        const input = screen.getByRole('textbox');
        expect(input).toHaveValue('сериал');
    });

    it('Snapshot', () => {
        const input = render(
            <InputSearch
                searchValue=''
                setSearchValue={onChange}
            />);
            
        expect(input).toMatchSnapshot();
    })
})