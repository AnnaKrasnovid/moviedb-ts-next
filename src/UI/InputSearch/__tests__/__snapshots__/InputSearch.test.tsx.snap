// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputSearch Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="input-box"
      >
        <div
          class="input-box__box "
        >
          <input
            autocomplete="off"
            class="input-box__input"
            id="search"
            minlength="1"
            name="search"
            placeholder="Поиск фильма"
            type="text"
            value=""
          />
          <span
            class="input-box__icon "
          />
        </div>
        <span
          class="error"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="input-box"
    >
      <div
        class="input-box__box "
      >
        <input
          autocomplete="off"
          class="input-box__input"
          id="search"
          minlength="1"
          name="search"
          placeholder="Поиск фильма"
          type="text"
          value=""
        />
        <span
          class="input-box__icon "
        />
      </div>
      <span
        class="error"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
